calcom/cal.com
# Fix early return when expanding RRULE in CalDav/iCal

## What does this PR do?



Recurring events in CalDav and iCal were only using the first instance of the event for availability checking. This fixes that issue. 

(might fix) #3589  (I'm not sure if this is the same issue or not, it looks to be a bit subtler)

No visual changes

**Environment**: Fixed against main.

## Type of change


## How should this be tested?

I looked for unit/integration tests but couldn't find any, so I might be missing some. To repro I created a new google calendar, accessed via CalDav. I created a single weekly recurring event in the calendar, and checked availability for each day in the month view for an event type in Cal.com. I could see that after my fix, the time slot with a clash was correctly not showing up in all 4 weeks, rather than just the first week.

## Checklist

I did all those things, I really think this is a pretty simple fix. Feel free to disagree though! 😊
